doc/adr/0005-deprecate-use-of-terragrunt.md

```diff
@@ -0,0 +1,50 @@
+# ADR-001. Record architectural decisions
+
+Date: 28/11/2019
+
+## Status
+
+Accepted
+
+## Context
+
+Historically the Digital Studio has made use of
+[Terragrunt](https://github.com/gruntwork-io/terragrunt) as a wrapper around
+terraform. The predominant reason is because the terraform code sturcture has
+been to split artchitecture up into seperate root modules and iterate over them
+using several invokations of terraform, this seems to be the gruntworks pattern.
+As these modules are invoked independantly of each other several configuration
+settings need to be kept consistent between these modules, which means repeating
+the same configuration in several files. Terragrunt's purpose is to abstract the
+backend and provider configurations out of terraform and into some external
+tfvars configuration that terragrunt reads and repeats it for you.
+
+## Decision
+
+Since the original release of terragrunt terraform has become much more mature.
+The backend and provider configs can now be determined by cli flags so having
+terragrunt is no longer necessary.
+
+Use of terragrunt complicates our terraform code, as new engineers need to be
+familiar with both terraform and terragrunt to be able to utilise our code.
+
+In addition to the above reasons, it is also been seen that our specific use
+of terragrunt has made extensive use of Bash scripts which have not been written
+in a portable manner. This has lead to issues where terragrunt has not been
+easily deployed from machines that do not meet the undocumented environment
+expected.
+
+## Consequences
+
+Terragrunt will be deprecated. Future projects the DSO team pick up will no
+longer make use of Terragrunt. If and when there is time, terragrunt may be
+removed from existing projects, however this is not work that should be actively
+pursued.
+
+Instead of relying on Terragrunt to orchastrate terraform modules, we will
+instead use a thin root module who's sole function is to call functional modules
+with correct inputs.
+
+## Credit
+
+Thanks to @daibach and the LAA team at MoJ. This first ADR and the repo structure was copied from https://github.com/ministryofjustice/laa-hosting-architectural-decisions.
```

Comment:
Right, yes we can. However I think it would be best done in a seperate ADR about terraform code structure. I was thinking of adding it into a seperate PR, but reflecting on it I think it may be better to have one PR for both ADR's and they can be linked togeather.

Does that sound reasonable?

Reply:
I don't mind, as long as we have a reference to it, I'm happy

Reply:
I've added in a new ADR for the terraform code structure. I'm not entirely sure it fully resolves your issue. I'm thinking of adding an examples directory to house code examples we may wish to refer to.